### PR TITLE
Fix route sorting & fix stage build

### PR DESCRIPTION
**Pull request by jonator**
## What is the purpose of the change

Route sorting would be lost when filtering candidate routes for unique pools.

### ClickUp Task

[ClickUp Task URL](PASTE_CLICKUP_TASK_URL_HERE)

### CODE REVIEW

File: packages/pools/src/router/routes.ts [hunk 1]
```diff
@@ -188,7 +188,7 @@ export class OptimizedRoutes implements TokenOutGivenInRouter {
         return includedRoutes;
       } else {
         route.pools.forEach(({ id }) => uniquePoolIds.add(id));
-        includedRoutes.unshift(route);
```
Comment (ValarDragon) [inline packages/pools/src/router/routes.ts:191]: LGTM to in isolation, though not seeing the corresponding sort line

Comment (jonator) [reply packages/pools/src/router/routes.ts:191]: Before they're filtered, they're sorted on line 182

File: packages/pools/src/router/__tests__/routes.spec.ts [hunk 2]
```diff
@@ -265,104 +265,6 @@ describe("OptimizedRoutes", () => {
       ]);
       expect(scaledOut.amount.toString()).toEqual(normalOut.amount.toString());
     });
-
-    it("uses preferred pool IDs - lifts preferred direct pool into used route", async () => {
```
Comment (jonator) [inline packages/pools/src/router/__tests__/routes.spec.ts:269]: No longer desired behavior

File: packages/pools/src/router/routes.ts (hunk 1 above)
Comment (ValarDragon) [reply packages/pools/src/router/routes.ts:191]: Hrmm, thats sorting by path length, not sorting by net price to user though?

**Review by MaxMillington — APPROVED** [review]

File: packages/pools/src/router/routes.ts (hunk 1 above)
Comment (jonator) [reply packages/pools/src/router/routes.ts:191]: Yeah it's just sorting by length. Price comparison will happen later

This PR contains a critical bug fix, we can follow up with improvements

**Pull request merged**